mongoengine/fields.py

```diff
@@ -104,8 +105,24 @@ def __init__(self, regex=None, max_length=None, min_length=None, **kwargs):
         self.regex = re.compile(regex) if regex else None
         self.max_length = max_length
         self.min_length = min_length
+        self._kwargs = kwargs
         super().__init__(**kwargs)
 
+    def __deepcopy__(self, memo):
+        memo[id(self)] = self
+
+        regex_bak = self.regex
+        self.regex = None
+
+        copied_obj = copy.deepcopy(self, memo)
```

Comment:
isn't this causing recursion to this same method?

Reply:
I'll try another approach. since `memo[id(self)] = self` is not a real fix for recursion issue